Mathlib/Data/List/Basic.lean

```diff
@@ -792,7 +792,7 @@ theorem getLast?_append_cons :
 #align list.last'_cons_cons List.getLast?_cons_cons
 
 theorem getLast?_append_of_ne_nil (l₁ : List α) :
-    ∀ {l₂ : List α} (_ : l₂ ≠ []), getLast? (l₁ ++ l₂) = getLast? l₂
+    ∀ {l₂ : List α}, l₂ ≠ [] → getLast? (l₁ ++ l₂) = getLast? l₂
```

Comment:
I'd argue that we should be naming these; they're part of the API of the lemma:
```suggestion
    ∀ {l₂ : List α} (hl₂ : l₂ ≠ []), getLast? (l₁ ++ l₂) = getLast? l₂
```
same for most of the list lemmas

Reply:
I reverted all changes to `Data/List`: Lean 4 won't allow us to use `hl₂` here, only `_hl₂`. Let's discuss this later.